portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,29 +59,89 @@
+
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      String userEmail = userService.getCurrentUser().getEmail();
+
+      json.addProperty("logoutUrl", logoutUrl);
+      json.addProperty("email", userEmail);
+      json.addProperty("name", name);
+    } else {
+      // Add a login link to the response if the user is not logged in.
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      json.addProperty("loginUrl", loginUrl);
+    }
+    
     // Query up to maxComments comment entities from Datastore with the user's specified sorting option.
+    int maxComments = getMaxCommentsToReturn(request);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = createCommentQuery(request);
     List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(maxComments));
 
-    // Construct a list of comments from the queried entities.
-    List<Comment> comments = results.stream().map((entity) -> {
-      long id = entity.getKey().getId();
-      String author = (String) entity.getProperty("author");
-      String commentText = (String) entity.getProperty("commentText");
-      long timestamp = (long) entity.getProperty("timestamp");
+    // Check if there are comments to translate.
+    if (results.isEmpty()) {
+      json.add("comments", new JsonArray());
+    } else {
+      // Construct a stream of comment texts from the queried entities.
+      Stream<String> commentTexts = results.stream().map((entity) -> {
+        String commentText = (String) entity.getProperty("commentText");
+        return commentText;
+      });
+
+      String languageCode = request.getParameter("language-code");
+      Stream<String> translatedCommentTexts = translateComments(commentTexts, languageCode);
+
+      // Construct a list of comments from the queried entities and translated comment texts.
+      List<Comment> comments = 
+          Streams.zip(results.stream(), translatedCommentTexts, (entity, translatedCommentText) -> {
+            long id = entity.getKey().getId();
+            String author = (String) entity.getProperty("author");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            return new Comment(id, author, translatedCommentText, timestamp);
+          }).collect(Collectors.toList());
+
+      // Convert the list of comments to a JsonElement.
+      JsonElement commentsJsonElement = convertToJsonElement(comments);
+      json.add("comments", commentsJsonElement);
+    }
 
-      return new Comment(id, author, commentText, timestamp);
-    }).collect(Collectors.toList());
+    // Send the JSON as the response.
+    response.setContentType("application/json; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+    response.getWriter().println(json.toString());
+  }
```

Comment:
This method is getting a bit large. Is there a good way to split it up? Perhaps split out the login vs comment logic?